### PR TITLE
fix: payload in bundle to be converted into an enum

### PR DESCRIPTION
**Pull request by b00f**
In this PR the payload replaced by message enum.

### CODE REVIEW

File: src/mvba/bundle.rs [hunk 1]
```diff
@@ -11,7 +11,14 @@ pub struct Bundle {
     /// This is the destination  module, it can be ABBA, VCBC or MVBA.
     pub module: String,
```
Comment (davidrusu) [inline src/mvba/bundle.rs:12]: The module string is no longer needed. We can match on the message enum to decide which module to dispatch the message to.

File: src/mvba/bundle.rs [hunk 2]
```diff
@@ -11,7 +11,14 @@ pub struct Bundle {
...
-    pub payload: Vec<u8>,
+    pub(crate) message: Message,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub enum Message {
+    VcbcMsg(vcbc::message::Message),
+    AbbaMsg(abba::message::Message),
+    MvbaMsg(mvba::message::Message),
 }
```
Comment (davidrusu) [inline src/mvba/bundle.rs:22]: The `*Msg` suffix is redundant. They are already prefixed by `Message::*`
```suggestion
pub enum Message {
    Vcbc(vcbc::message::Message),
    Abba(abba::message::Message),
    Mvba(mvba::message::Message),
}
```

File: src/mvba/consensus.rs [hunk 3]
```diff
@@ -88,8 +89,13 @@ impl Consensus {
             vcbc::MODULE_NAME => match bundle.target {
                 Some(target) => match self.vcbc_map.get_mut(&target) {
                     Some(vcbc) => {
-                        let msg = bincode::deserialize(&bundle.payload)?;
-                        vcbc.receive_message(bundle.initiator, msg, &mut self.broadcaster)?;
+                        let msg = match &bundle.message {
+                            VcbcMsg(msg) => msg,
+                            _ => {
+                                return Err(Error::InvalidMessage("unexpected message".to_string()))
```
Comment (davidrusu) [inline src/mvba/consensus.rs:95]: Yeah, if you pattern match on `bundle.message` instead of `bundle.module`, you no longer have this error checking

Comment (b00f) [reply src/mvba/consensus.rs]: True, I forgot it. Thanks for reminding

**Review by davidrusu — APPROVED** [review]

**Pull request merged**